### PR TITLE
docs(routing): add middleware section

### LINKED ISSUE #1306: Missing middlewares docs
### Environment

/

### Reproduction

/

### Describe the bug

/

### Additional context

_No response_

### Logs

_No response_

### PR DESCRIPTION
**Pull request by Barbapapazes**
### 🔗 Linked issue

fix #1306

### ❓ Type of change

- [x] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Usage of middlewares are not documented. This PR is a try to add documentation about middlewares.

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: docs/content/1.guide/4.middlewares.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+# Middlewares
+
+Nitro supports globals middlewares to hook into the request lifecycle.
+
+::alert{type=primary}
+Middlewares can only interact before the request is processed.
+::
+
+Middlewares are auto-registered withing the `middlewares/` directory.
```
Comment (Barbapapazes) [inline docs/content/1.guide/4.middlewares.md:14]:
```suggestion
Middlewares are auto-registered within the `middlewares/` directory.
```

File: docs/content/1.guide/4.middlewares.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+# Middlewares
+
+Nitro supports globals middlewares to hook into the request lifecycle.
+
+::alert{type=primary}
+Middlewares can only interact before the request is processed.
+::
+
+Middlewares are auto-registered within the `middlewares/` directory.
```
Comment (Barbapapazes) [inline docs/content/1.guide/4.middlewares.md:14]: folder name does not end with a s

Comment (Barbapapazes) [reply docs/content/1.guide/4.middlewares.md:14]: need to be confirmed by seeing code.

**Review by Hebilicious — APPROVED** [review]

**Comment by Hebilicious** [comment]
This is looking good, thanks @Barbapapazes !

**Review by pi0 — APPROVED** [review]
Thanks! For now added it under routing section for similification

**Pull request merged**